packages/material-ui-styled-engine-sc/src/StylesProvider/StylesProvider.js

```diff
@@ -7,22 +7,14 @@ let injectFirstNode;
 export function StylesProvider(props) {
   const { injectFirst, children } = props;
 
-  React.useEffect(() => {
-    if (injectFirst && !injectFirstNode) {
-      const head = document.head;
-      injectFirstNode = document.createElement('style');
-      injectFirstNode.setAttribute('data-styled', 'active');
-      head.insertBefore(injectFirstNode, head.firstChild);
-    }
+  if (injectFirst && !injectFirstNode && typeof window !== 'undefined') {
```

Comment:
Something is still not working as it should here. The `styled-components` style tag is injected fist only after switching pages. 😕